src/lib/tls/tls_callbacks.h

```diff
@@ -463,6 +464,33 @@ class BOTAN_PUBLIC_API(2, 0) Callbacks /* NOLINT(*-special-member-functions) */
       virtual std::unique_ptr<PK_Key_Agreement_Key> tls_generate_ephemeral_key(
          const std::variant<TLS::Group_Params, DL_Group>& group, RandomNumberGenerator& rng);
 
+      /**
+       * Generate an ECDH, X25519, or X448 key pair for the TLS 1.2 handshake.
+       *
+       * Note that this callback is called exclusively by TLS 1.2 to handle the
+       * public key serialization format explicitly. TLS 1.3 fixes this format
+       * to 'uncompressed' and does not allow negotiating anything else.
+       *
+       * Users may override this if they want to provide a custom keypair type
+       * to offload TLS 1.2's ECDH handling to custom hardware, for instance.
+       *
+       * Typical use cases of the library don't need to do that and serious
+       * security risks are associated with customizing TLS's key exchange
+       * mechanism.
+       *
+       * @throws TLS_Exception(Alert::DecodeError) if the @p group is not known.
+       *
+       * @param group the group identifier to generate an ephemeral keypair for
+       *              TLS 1.2 allows for specifying custom discrete logarithm
+       *              parameters as part of the protocol. Hence the variant<>.
+       * @param rng a random number generator
+       * @param tls12_ecc_pubkey_encoding_format the key's serialization format
+       *
+       * @return an ECDH private key of an algorithm usable for key agreement
+       */
+      virtual std::unique_ptr<PK_Key_Agreement_Key> tls_generate_ephemeral_ecdh_key(
```

Comment:
When it's only meant for TLS1.2, we could clarify that in the function name as well (e.g. `tls12_generate_...`)?

Reply:
Fair point.